### ISSUE TITLE
Duplicate crontab expressions when conf.d folder is used

### ISSUE THREAD

**Issue by johnatank** [offen/docker-volume-backup#139]
Hi team,

Looks like passing `crontab -l` as input for crontab in entrypoint.sh leads to duplicates on container restart. Duplicated cron expressions run duplicated backups (+ run-parts does nothing and is spamming the log):
- `crontab -l` output after container was restarted 5 times:
```
08:48:23 username@host tmp → docker exec test crontab -l
# do daily/weekly/monthly maintenance
# min   hour    day     month   weekday command
*/15    *       *       *       *       run-parts /etc/periodic/15min
0       *       *       *       *       run-parts /etc/periodic/hourly
0       2       *       *       *       run-parts /etc/periodic/daily
0       3       *       *       6       run-parts /etc/periodic/weekly
0       5       1       *       *       run-parts /etc/periodic/monthly

0 9 * * * /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
0 9 * * * /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
0 9 * * * /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
0 9 * * * /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
0 9 * * * /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
0 9 * * * /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
```
- rules were executed 6 times:
```
crond: USER root pid  17 cmd run-parts /etc/periodic/15min
crond: USER root pid  18 cmd run-parts /etc/periodic/hourly
crond: USER root pid  19 cmd /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
crond: USER root pid  20 cmd /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
crond: USER root pid  21 cmd /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
crond: USER root pid  22 cmd /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
crond: USER root pid  23 cmd /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
crond: USER root pid  24 cmd /bin/sh -c 'set -a; source /etc/dockervolumebackup/conf.d/daily.env; set +a && backup' 2>&1
```
Removing `crontab -l` from this will fail to load multiple .env files in conf.d folder to crontab.

I think better option would be to create a file with list of rules and then pass that file to crontab, smth like that:
```
else
  echo "/etc/dockervolumebackup/conf.d was found, using configuration files from this directory."
  touch /tmp/rules.txt

  for file in /etc/dockervolumebackup/conf.d/*; do
    source $file
    BACKUP_CRON_EXPRESSION="${BACKUP_CRON_EXPRESSION:-@daily}"
    echo "Appending cron.d entry with expression $BACKUP_CRON_EXPRESSION and configuration file $file"
    (echo "$BACKUP_CRON_EXPRESSION /bin/sh -c 'set -a; source $file; set +a && backup' 2>&1") >> /tmp/rules.txt
  done
  crontab /tmp/rules.txt
  rm /tmp/rules.txt
fi
```
Doubt above is the best code. Alternatively, I guess /etc/crontabs/root can be modified.

Thanks,
Maksym

**Comment by m90** [comment]
Oh, that's a good catch. I think there should be a simpler solution though: before anything happens in the `entrypoint.sh` script, we could run `crontab -r`, effectively wiping all existing cron jobs, creating a green field for the script to add the jobs in the way it is done now. Can you think of any reason that would not work?

**Comment by m90** [comment]
This is fixed in v2.21.0
